src/macaron/policy_engine/__main__.py

```diff
@@ -0,0 +1,187 @@
+    if not os.path.isfile(database_path):
+        logger.error("Unable to open database %s", database_path)
+        sys.exit(1)
+
+    metadata = MetaData()
+    engine = create_engine(f"sqlite:///{database_path}", echo=False)
+    metadata.reflect(engine)
+
+    prelude = get_souffle_import_prelude(os.path.abspath(database_path), metadata)
+
+    for table_name in metadata.tables.keys():
+        table = metadata.tables[table_name]
+        if table_name[0] == "_":
+            prelude.update(project_table_to_key(f"{table_name[1:]}_attribute", table))
+            prelude.update(project_with_fk_join(table))
+
+    return prelude
+
+
+def copy_prelude(database_path: os.PathLike | str, sfl: SouffleWrapper, prelude: SouffleProgram | None = None) -> None:
+    """
+    Generate and copy the prelude into the souffle instance's include directory.
+
+    Parameters
+    ----------
+    database_path: os.PathLike | str
+        The path to the database the facts will be imported from
+    sfl: SouffleWrapper
+        The souffle execution context object
+    prelude: SouffleProgram | None
+        Optional, the prelude to use for the souffle program, if none is given the default prelude is generated from
+        the database at database_path.
+    """
+    if prelude is None:
+        prelude = get_generated(database_path)
+    sfl.copy_to_includes("import_data.dl", str(prelude))
+
+    folder = os.path.join(os.path.dirname(__file__), "prelude")
+    for file_name in os.listdir(folder):
+        full_file_name = os.path.join(folder, file_name)
+        if not os.path.isfile(full_file_name):
+            continue
+        with open(full_file_name, encoding="utf-8") as file:
+            text = file.read()
+            sfl.copy_to_includes(file_name, text)
+
+
+def policy_engine(database_path: str, policy_file: str) -> dict:
+    """Invoke souffle and report result."""
+    with SouffleWrapper() as sfl:
+        copy_prelude(database_path, sfl)
+        with open(policy_file, encoding="utf-8") as file:
+            text = file.read()
+
+        try:
+            res = sfl.interpret_text(text)
+        except SouffleError as error:
+            logger.error("COMMAND: %s", error.command)
+            logger.error("ERROR: %s", error.message)
+            sys.exit(1)
+
+        return res
+
+
+def non_interactive(database_path: str, show_prelude: bool, policy_file: str) -> bool:
+    """Evaluate a policy based on configuration and exit."""
+    if show_prelude:
+        prelude = get_generated(database_path)
+        logger.info("\n%s", prelude)
+        return False
+
+    res = policy_engine(database_path, policy_file)
+
+    output = []
+    for key, values in res.items():
+        output.append(str(key))
+        for value in values:
+            output.append(f"    {value}")
+
+    logger.info("Policy results:\n%s", "\n".join(output))
+
+    return any(res["failed_policies"])
+
+
+def main() -> Never:
```

Comment:
Can you please add this as an entrypoint script in [`pyproject.toml`](https://github.com/oracle-samples/macaron/pull/46/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R47)?

Reply:
Done in https://github.com/oracle-samples/macaron/pull/46/commits/f73924020c8f8c3b629f31dcc4700d311450bd02